twenty-first/src/math/tip5.rs

```diff
@@ -673,10 +675,39 @@ impl Sponge for Tip5 {
     }
 }
 
+impl Hasher for Tip5 {
+    fn finish(&self) -> u64 {
+        self.state[0].value()
+    }
+
+    fn write(&mut self, bytes: &[u8]) {
+        let bfield_elements = bytes.chunks(BFieldElement::BYTES).map(|chunk| {
+            let mut buffer = [0u8; BFieldElement::BYTES];
+            buffer[..chunk.len()].copy_from_slice(chunk);
+            BFieldElement::new(u64::from_le_bytes(buffer))
+        });
+
+        let binding: Vec<BFieldElement> = bfield_elements.collect();
+        let bfield_absorb_chunks = binding.chunks(Tip5::RATE);
+
+        bfield_absorb_chunks.for_each(|absorb_rate_chunk| {
+            let mut asborb_chunk_buffer: [BFieldElement; Tip5::RATE] =
+                [BFieldElement::new(0); Tip5::RATE];
+            for (i, bfield_elt) in absorb_rate_chunk.iter().enumerate() {
+                asborb_chunk_buffer[i] = *bfield_elt;
+            }
```

Comment:
Personally, I like to avoid explicit indexing when I can.

```suggestion
            for (buffer_elt, chunk_elt) in buffer.iter_mut().zip(chunk) {
                *buffer_elt = chunk_elt
            }
```

This one also boils down to personal preference; feel free to ignore this suggestion.